### PR TITLE
Fix request id key

### PR DESCRIPTION
**Pull request by burov**
Fix request id key to conform with https://en.wikipedia.org/wiki/List_of_HTTP_header_fields#Common_non-standard_request_fields

### CODE REVIEW

File: requestid/requestid.go [hunk 1]
```diff
@@ -14,7 +14,8 @@ import (
 // DefaultRequestIDKey is the metadata key name for request ID
 const (
 	DeprecatedRequestIDKey = "Request-Id"
-	DefaultRequestIDKey    = "request_id"
+	DefaultRequestIDKey    = "X-Request-ID"
```
Comment (kd7lxl) [inline requestid/requestid.go:17]: Note that this is already implemented in the ingress-controller:
https://github.com/Infoblox-CTO/athena.devops/blob/d9556173c9d749f86515876cadeb6fa0ec9a60d8/cluster-provisioning/charts/base/files/nginx.tmpl#L322-L329

But not forwarded to identity:
https://github.com/Infoblox-CTO/athena.devops/blob/d9556173c9d749f86515876cadeb6fa0ec9a60d8/cluster-provisioning/charts/base/files/nginx.tmpl#L920
(on auth requests, it's set as `Request-ID` only)

File: logging/interceptor_test.go [hunk 2]
```diff
@@ -38,7 +39,7 @@ var (
 	buf         bytes.Buffer
 	reader      io.Reader
 	testLogger  = New("Info")
-	testMD      = metautils.NiceMD{}.Set(testAuthorizationHeader, testJWT).Set(DefaultRequestIDKey, testRequestID).Set(testCustomHeaderKey, testCustomHeaderVal)
+	testMD      = metautils.NiceMD{}.Set(testAuthorizationHeader, testJWT).Set(requestid.DefaultRequestIDKey, testRequestID).Set(testCustomHeaderKey, testCustomHeaderVal)
```
Comment (kd7lxl) [inline logging/interceptor_test.go:42]: Please add a case that looks for `DeprecatedRequestIDKey` so we can verify backwards compatibility.

File: requestid/requestid.go (hunk 1 above)
Comment (Evgeniy-L) [reply requestid/requestid.go:17]: @kd7lxl, please avoid references to data from private repos in public repositories

Comment (burov) [reply requestid/requestid.go:17]: @kd7lxl We probably can forward it, it a good point

File: logging/interceptor_test.go (hunk 2 above)
Comment (burov) [reply logging/interceptor_test.go:42]: Good point, did it

File: requestid/requestid.go (hunk 1 above)
Comment (drewwells) [reply requestid/requestid.go:17]: broke authz middleware as well https://github.com/Infoblox-CTO/ngp.authz.middleware/pull/23/files#diff-320a504f903c9c37cd1c4852412110f8R216

**Pull request merged**